pkg/netpol/eval/exposure.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package eval
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/np-guard/netpol-analyzer/pkg/netpol/eval/internal/k8s"
+	"github.com/np-guard/netpol-analyzer/pkg/netpol/internal/common"
+)
+
+// this file contains eval.PolicyEngine funcs which are related to exposure-analysis feature
+
+// addPotentiallyExposedPods adds pods which are potentially exposed to connect with (from/to) existing pods in the resources
+func (pe *PolicyEngine) addPotentiallyExposedPods() error {
+	// first adds a pod that represent connections with any namespace
+	_, err := pe.AddPodByNameAndNamespace(common.PodInExposedNs, common.AllNamespaces, nil)
+	if err != nil {
+		return err
+	}
+	// scan policies's rules : generate fake-exposure pod for rules with no match in the resources
+	for _, nsNetpolsMap := range pe.netpolsMap {
```

Comment:
This requires iterating all policies, and for each policy iterate all selectors, and for each selector iterate all namespaces (or pods in case of pod selectors) , so that we add "representative pods" for such empty selectors... 
Can we do something more efficient? 
e.g. add in this stage a "representative pod" for each selector (without checking if it matched by input workload/ns or not), and during the analysis save on-the-fly if selectors are matched (by actual pods, not 'fake' ones), and at the end keep the results of the analysis only for representative pods associated with selectors that were not matched by any actual pod? 
Try to think if this or other possible optimization ideas are possible, this can be a separate stage in another sub-task, but we should consider the order of sub-tasks..

Reply:
handled in #314